### PR TITLE
Move request package into public/shared folder

### PR DESCRIPTION
**Pull request by fmartingr**
#### Summary

From #24408

Move the `request` package into `public/shared` (from `channels/app`).

#### Ticket Link
--

#### Release Note
```release-note
Move `request` package into the public shared folder
```

### CODE REVIEW

**Comment by agnivade** [comment]
Continuing our conversation from:https://github.com/mattermost/mattermost/pull/24408/files#r1310068962, I realized the problem, we are removing the `app/request` package. Instead adding a new one. That's why it's still taking the reference from the older one. I meant to "move" the package under public/shared. Let's do that.

**Review by agnivade — CHANGES_REQUESTED** [review]
Left a comment

**Comment by fmartingr** [comment]
> Continuing our conversation from:https://github.com/mattermost/mattermost/pull/24408/files#r1310068962, I realized the problem, we are removing the `app/request` package. Instead adding a new one. That's why it's still taking the reference from the older one. I meant to "move" the package under public/shared. Let's do that.

It seems I didn't notice and I left the original package in there. https://github.com/mattermost/mattermost/pull/24420/commits/f156dc7e2fe74e2cdfa58e074d0376b58c4ca08d

**Review by agnivade — APPROVED** [review]
Nice! Looks great now

**Comment by fmartingr** [comment]
> Nice! Looks great now

@agnivade Since @jespino is out, can you assign a second reviewer?

File: server/channels/app/layer_generators/opentracing_layer.go.tmpl [hunk 1]
```diff
@@ -140,4 +141,4 @@ func (a *{{.Name}}) Timezones() *timezones.Timezones {
 }
 func (a *{{.Name}}) SetServer(srv *app.Server) {
        a.srv = srv
-}
\ No newline at end of file
+}
```
Comment (hanzei) [inline server/channels/app/layer_generators/opentracing_layer.go.tmpl:144]: Is the generated file related to `server/channels/app/layer_generators/opentracing_layer.go.tmpl` already up to date?

**Review by hanzei — CHANGES_REQUESTED** [review]

File: server/channels/app/layer_generators/opentracing_layer.go.tmpl (hunk 1 above)
Comment (fmartingr) [reply server/channels/app/layer_generators/opentracing_layer.go.tmpl:144]: I've run `make generated` which I though it ran everything that should be run. Did I miss anything?

**Review by hanzei — APPROVED** [review]
LGTM :+1:

**Pull request merged**